astropy/io/fits/tests/test_image.py

```diff
@@ -1777,6 +1778,37 @@ def test_compressed_header_missing_znaxis(self):
         with pytest.raises(TypeError):
             comp_hdu.compressed_data
 
+    @pytest.mark.parametrize(('keyword', 'dtype', 'expected'),
+                             [(kk, dd, ee) for kk, (dd, ee) in
+                             itertools.product(["BSCALE", "BZERO"],
+                                 zip([np.uint8, np.int16, np.int32],
+                                     [np.float32, np.float32, np.float64]))])
+    def test_compressed_type(self, keyword, dtype, expected):
+        """
+        If BSCALE,BZERO is set to floating point values, the image
+        should be floating-point.
+
+        Second part of https://github.com/astropy/astropy/pull/6486
+
+        Parameters
+        ----------
+        keyword : `str`
+            Keyword to set to a floating-point value to trigger
+            floating-point pixels.
+        dtype : `numpy.dtype`
+            Type of original array.
+        expected : `numpy.dtype`
+            Expected type of uncompressed array.
+        """
+        value = 1.23345  # A floating-point value
+        hdu = fits.CompImageHDU(np.arange(0, 10, dtype=dtype))
+        hdu.header[keyword] = value
+        hdu.writeto(self.temp('test.fits'))
+        del hdu
```

Comment:
is there a reason why you have to delete it here? Just curious. :)

Reply:
I use `hdu` in the next line, and just wanted to be clear it's not this `hdu`.